Homework4/Homework4/WeeklyTaskService.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework4
+{
+    internal class WeeklyTaskService
+    {
+        private readonly List<WeeklyTask> tasks;
+
+        public WeeklyTaskService()
+        {
+            tasks = new();
+        }
+
+        public void AddNewTask()
+        { 
+            Console.WriteLine("Add task in format: {},{},{},{}");
+            var inputData = Console.ReadLine();
+            var task = ParseNewTask(inputData);
+            AddNewTask(task);
+        }
+
+        public void UpdateTask()
+        {
+            Console.WriteLine("Input number to update:");
+            var inputNumber = Console.ReadLine();
+            var taskNumber = int.Parse(inputNumber);
+            
+            Console.WriteLine("Input new task data:");
+            var inputTaskData = Console.ReadLine();
+            var task = ParseNewTask(inputTaskData);
+            tasks[taskNumber - 1] = task;
+        }
+
+        public void FilterTaskByPriority()
+        {
+            Console.WriteLine("Input priority:");
+            var priority = Console.ReadLine();
+
+            for (int i = 0; i < tasks.Count; i++)
+            {
+                var task = tasks[i];
+
+                if (((PriorityTask) tasks[i]).GetPriority().ToString() == priority)
```

Comment:
hm, it should fail if I create any non priority task
this cast `(PriorityTask) tasks[i]` is not safe (it might throw) - you not to ensure that the `task[i]` is `PriorityTask`.
Use `is` or `as` operators to ensure the type safety.
```suggestion
                if (tasks[i] is PriorityTask priorityTask)
```